### PR TITLE
chore: update cSpell globs and fix typos

### PR DESCRIPTION
**Pull request by nschonni**
**What kind of change does this PR introduce?**

**Did you add tests for your changes?**

**Does this PR introduce a breaking change?**

**What needs to be documented once your changes are merged?**

### CODE REVIEW

File: declarations.d.ts [hunk 1]
```diff
@@ -204,7 +204,7 @@ declare module "@webassemblyjs/ast" {
 		raw?: string
 	): FloatLiteral;
 	export function global(globalType: string, nodes: Node[]): Global;
-	export function identifier(indentifier: string): Identifier;
+	export function identifier(identifier: string): Identifier;
```
Comment (nschonni) [inline declarations.d.ts:207]: Unsure if this counts as a breaking change or not

File: package.json [hunk 2]
```diff
@@ -159,7 +159,7 @@
     "type-lint": "tsc",
     "typings-test": "tsc -p tsconfig.types.test.json",
     "module-typings-test": "tsc -p tsconfig.module.test.json",
-    "spellcheck": "cspell \"{.github,benchmark,bin,examples,hot,lib,schemas,setup,tooling}/**/*.{md,yml,yaml,js,json}\" \"*.md\"",
+    "spellcheck": "cspell \"**/*.*\"",
```
Comment (nschonni) [inline package.json:162]: Moved ignores to the `cspell.json` file

File: cspell.json [hunk 3]
```diff
@@ -1,241 +1,292 @@
...
     "examples/**/README.md",
-    "**/webpack.lock.data/**"
+    "**/webpack.lock.data/**",
+    "package.json",
+    "**/**/*.snap",
+    "test/cases/json/weird-properties/globals.json",
+    "**/*.svg",
+    "**/*.wasm",
+    "test/**/module.js",
+    "test/**"
```
Comment (nschonni) [inline cspell.json:290]: With this re-enabled, there is still `Issues found: 279 in 67 files`

File: types.d.ts [hunk 4]
```diff
@@ -11634,7 +11634,7 @@ declare interface UserResolveOptions {
 	restrictions?: (string | RegExp)[];
 
 	/**
-	 * Use only the sync constiants of the file system calls
+	 * Use only the sync constraints of the file system calls
```
Comment (nschonni) [inline types.d.ts:11637]: This appears to be an upstream issue in https://github.com/webpack/enhanced-resolve, so i'll submit a PR there

**Review by sokra — APPROVED** [review]

**Comment by nschonni** [comment]
Thanks for the review @sokra! Looks like the GitHub Actions might still need to get approved to finish the green checkmark

**Comment by sokra** [comment]
Thanks

**Pull request merged**